app/utility/api_docs.py

```diff
@@ -0,0 +1,171 @@
+import copy
+from collections import defaultdict
+
+
+class Responses:
+    DEFAULT_RESPONSE = {
+        '200': {
+            'description': 'successful operation'
+        },
+        '400': {
+            'description': 'invalid request'
+        },
+        '500': {
+            'description': 'internal server error'
+        }
+    }
+    JSON_RESPONSE = copy.copy(DEFAULT_RESPONSE)
+    JSON_RESPONSE['200']['content'] = 'application/json'
+
+
+class Schemas:
+    AGENT = dict(Agent=dict(type='object', description='Agent', properties=dict(
```

Comment:
is this possible to derive from the agent object, same with operation, so that it is dynamic

Reply:
I tried pretty hard to get it that to work, but there just isn't quite enough information.  The approach I went with was inspecting the args/kwargs from __init__ and the display functions.   I may have been able to get the field names, but wasn't able to infer the types.